frontend/src/routes/manage/shared.tsx

```diff
@@ -0,0 +1,575 @@
+                    ":focus-within": {
+                        borderRadius: 4,
+                        outline: `2.5px solid ${COLORS.focus}`,
+                    },
+                }}>{title}</div>
+                {syncInfo && !syncInfo.isSynced && (
+                    <span css={{
+                        padding: "0 8px",
+                        fontSize: "small",
+                        borderRadius: 10,
+                        backgroundColor: COLORS.neutral10,
+                    }}>
+                        {t(syncInfo.notReadyLabel)}
+                    </span>
+                )}
+            </div>
+            {description}
+        </td>
+        {customColumns}
+    </tr>;
+};
+
+type ColumnHeaderProps = {
+    label: string;
+    sortKey: SortColumn;
+    vars: ItemVars;
+};
+
+const ColumnHeader: React.FC<ColumnHeaderProps> = ({ label, sortKey, vars }) => {
+    const { t } = useTranslation();
+    const direction = vars.order.direction === "ASCENDING" ? "DESCENDING" : "ASCENDING";
+    const directionTransKey = direction === "ASCENDING" ? "ascending" : "descending";
+
+    return <th>
+        <Link
+            aria-label={
+                t("manage.item-table.columns.description", {
+                    title: label,
+                    direction: t(`manage.item-table.columns.${directionTransKey}`),
+                })
+            }
+            to={varsToLink({
+                order: {
+                    column: sortKey,
+                    direction,
+                },
+                limit: vars.limit ?? LIMIT,
+                offset: vars.offset ?? 0,
+            })}
+            css={{
+                display: "inline-flex",
+                alignItems: "center",
+                cursor: "pointer",
+                transition: "color 70ms",
+                textDecoration: "none",
+                borderRadius: 4,
+                outlineOffset: 1,
+                "& > svg": {
+                    marginLeft: 6,
+                    fontSize: 22,
+                },
+            }}
+        >
+            {label}
+            {vars.order.column === sortKey && match(vars.order.direction, {
+                // Seems like this is flipped right? But no, a short internal
+                // poll showed that this matches the intuition of almost everyone.
+                "ASCENDING": () => <LuArrowDownNarrowWide />,
+                "DESCENDING": () => <LuArrowUpWideNarrow />,
+            }, () => null)}
+        </Link>
+    </th>;
+};
+
+const PageNavigation: React.FC<SharedProps> = ({ connection, vars }) => {
+    const { t } = useTranslation();
+    const pageInfo = connection.pageInfo;
+    const total = connection.totalCount;
+
+    const limit = vars.limit ?? LIMIT;
+    const offset = vars.offset ?? 0;
+
+    const prevOffset = Math.max(0, offset - limit);
+    const nextOffset = offset + limit;
+    const lastOffset = total > 0
+        ? Math.floor((total - 1) / limit) * limit
+        : 0;
+
```

Comment:
Seems a bit roundabout to me. Maybe thats a remnant from the previous code but why not just pass the current `page` to this component. Then you don't have to do as much calculation and stuff. I suppose the whole frontend hardly needs to know about `offset`. The offset only needs to be calculated temporarily to send to the API, but I think everywhere else I would just use `page`. So that also means not passing these GraphQL vars to `<ManageItems />`, but just the "query vars", i.e. `page`, sort order and potentially the `limit` if we really want to keep that dynamic.